Xamarin.Essentials/Platform/Platform.android.cs

```diff
@@ -152,4 +159,95 @@ void Application.IActivityLifecycleCallbacks.OnActivityStopped(Activity activity
         {
         }
     }
+
+    [Activity(ConfigurationChanges = ConfigChanges.Orientation | ConfigChanges.ScreenSize)]
+    class IntermediateActivity : Activity
```

Comment:
This intermediate activity is totally generic and can be used for any sort of external activity. Not only that, it will always ensure that the tasks will complete. The only exception is if the ENTIRE app gets killed - in this case, nothing matters as all managed state is lost and the tasks will have been cleared.